src/types.ts

```diff
@@ -712,9 +712,17 @@ export interface FSXAApi {
   fetchElement: <T = Page | GCAPage | Dataset | Image | any | null>(
     params: FetchElementParams
   ) => Promise<T>
-  fetchByFilter: (params: FetchByFilterParams) => Promise<any>
+  fetchByFilter: (params: FetchByFilterParams) => Promise<FetchResponse>
   fetchNavigation: (params: FetchNavigationParams) => Promise<NavigationData | null>
   fetchProjectProperties: (
     params: FetchProjectPropertiesParams
   ) => Promise<Record<string, any> | null>
 }
+
+export interface FetchResponse {
+  page: number
+  pagesize: number
+  totalPages?: number
+  size?: number
+  items: unknown[]
```

Comment:
feedback:
make FetchResponse generic. 

that would switch the method signature from fetchByfilter to this:

```function fetchByFilter<T = unknown>(): FetchResponse<T>```

on PWA Side, developers can then provide the type they're expecting to get.

Reply:
I like that. Will change it.

Reply:
I added that. Nevertheless one should be aware that this is still a response that could in theory contain anything. So just adding the generic type won't replace proper validation of the received data.

Reply:
After discussing it, we changed our mind. Instead of generics we use castings so the user won't think the returned data will definitely be of that type.